src/preamble.js

```diff
@@ -1056,45 +1040,71 @@ function createWasm(env) {
     receiveInstance(output['instance']);
 #endif
   }
+
   function instantiateArrayBuffer(receiver) {
-    getBinaryPromise().then(function(binary) {
+    return getBinaryPromise().then(function(binary) {
       return WebAssembly.instantiate(binary, info);
     }).then(receiver, function(reason) {
       err('failed to asynchronously prepare wasm: ' + reason);
       abort(reason);
-    });
+    })
```

Comment:
our convention is to have `;` at the end of lines (minifiers can remove it later etc.)